runtime/v1/shim/service.go

```diff
@@ -503,47 +503,46 @@ func (s *Service) processExits() {
 	}
 }
 
-func (s *Service) allProcesses() []process.Process {
+func (s *Service) getProcess(pid int) process.Process {
 	s.mu.Lock()
 	defer s.mu.Unlock()
 
-	res := make([]process.Process, 0, len(s.processes))
 	for _, p := range s.processes {
-		res = append(res, p)
+		if p.Pid() == pid {
```

Comment:
thanks for contribution about this. It can save memory here. 
Could you mind to update `runtime/v2/runc/service.go` in this patch? thanks

Reply:
There are two directories under runtime/v2/runc : v1 and v2.
I looked at the service.go under those two directories and found:
```
runtime/v2/runc/v1/service.go:
func (s *service) getProcess(execID string) (process.Process, error) {
```
where single Process is returned.

Is there other func which I should examine ?

Reply:
Both. But the function `getProcess(execID string)` is used to get process by ID instead of pid. need to choose other name to handle this. :)

Reply:
I think your suggestion is to keep method names across service.go files consistent.

How about naming the new method in runtime/v1/shim/service.go getProcessByPid ?

Reply:
For this PR, there is no need to introduce getProcess() since there is only one caller.

I have refactored the code.

@fuweid 
Can you take another look ?

Reply:
@fuweid 
Please review again.